src/main/java/uk/gov/hmcts/darts/cases/service/impl/CaseServiceImpl.java

```diff
@@ -130,5 +133,18 @@ public List<AdvancedSearchResult> advancedSearch(GetCasesSearchRequest request)
         return AdvancedSearchResponseMapper.mapResponse(hearings);
     }
 
-
+    public List<EventResponse> getEvents(Integer caseId, Integer hearingId) {
+        List<Hearing> hearings = getCaseHearings(caseId);
+        if (hearings.isEmpty()) {
+            throw new DartsApiException(CaseApiError.CASE_NOT_FOUND);
+        }
+        Hearing hearing = hearings
+            .stream()
+            .filter(hearingEntity -> hearingEntity.getId().equals(hearingId))
```

Comment:
It feels like we should probably add a new finder service here?

Reply:
This has been simplified since we don't need case Id.